Tasks/Maven/maventask.ts

```diff
@@ -202,9 +202,6 @@ async function execBuild() {
             if (userRunFailed || codeAnalysisFailed) {
                 tl.setResult(tl.TaskResult.Failed, "Build failed."); // Set task failure
             }
-            else {
-                tl.setResult(tl.TaskResult.Succeeded, "Build Succeeded."); // Set task success
-            }
```

Comment:
This is the core of the fix - the booleans userRunFailed and codeAnalysisFailed are only set when there are errors in the associated sections. This does not account for when the build breaker fails the build, which is not an error but a deliberate act, so no errors are thrown and we instead enter the "else" statement which sets the task result back to "Succeeded".

Previously this was not an issue because calling setResult with "Failed" immediately exited the process, but this is no longer the case.